salt/minion.py

```diff
@@ -3433,13 +3434,8 @@ def _return_pub_syndic(self, values, master_id=None):
             future, data = self.pub_futures.get(master, (None, None))
             if future is not None:
                 if not future.done():
-                    if master == master_id:
-                        # Targeted master previous send not done yet, call again later
-                        return False
```

Comment:
The idea of this block is: if syndic still has unsent data for the targeted master at this moment it means we got back here too fast and the data wasn't sent yet.
Full removal of this block would make syndic sending data randomly to different masters when syndic is on load. I'd better suggest to write good tests for the particular case you're trying to resolve first to have a way to check the fix works in the correct way.